ocs_ci/ocs/external_ceph.py

```diff
@@ -0,0 +1,924 @@
+            return [node for node in self.node_list if node.role == role]
+        elif ignore:
+            return [node for node in self.node_list if node.role != ignore]
+        else:
+            return list(self.node_list)
+
+    def get_ceph_objects(self, role=None):
+        """
+        Get Ceph Object by role. Returns all objects if role is not defined. Ceph object can be Ceph demon, client,
+        installer or generic entity. Pool role is never assigned to Ceph object and means that node has no Ceph objects
+
+        Args:
+            role (str): Ceph object's role as str
+
+        Returns:
+            list: ceph objects
+
+        """
+        node_list = self.get_nodes(role)
+        ceph_object_list = []
+        for node in node_list:
+            ceph_object_list.extend(node.get_ceph_objects(role))
+        return ceph_object_list
+
+    def get_ceph_object(self, role, order_id=0):
+        """
+        Returns single ceph object. If order id is provided returns that occurrence from results list, otherwise returns
+        first occurrence
+
+        Args:
+            role(str): Ceph object's role
+            order_id(int): order number of the ceph object
+
+        Returns:
+            CephObject: ceph object
+
+        """
+        try:
+            return self.get_ceph_objects(role)[order_id]
+        except IndexError:
+            return None
+
+    def get_ceph_demons(self, role=None):
+        """
+        Get Ceph demons list
+
+        Returns:
+            list: list of CephDemon
+
+        """
+        node_list = self.get_nodes(role)
+        ceph_demon_list = []
+        for node in node_list:  # type: CephNode
+            ceph_demon_list.extend(node.get_ceph_demons(role))
+        return ceph_demon_list
+
+    @property
+    def ceph_demon_stat(self):
+        """
+        Retrieves expected numbers for demons of each role
+
+        Returns:
+            dict: Ceph demon stats
+
+        """
+        ceph_demon_counter = {}
+        for demon in self.get_ceph_demons():
+            if demon.role == 'mgr':
+                continue
+            increment = 1  # len(self.get_osd_devices(demon.node)) if demon.role == 'osd' else 1
+            ceph_demon_counter[demon.role] = ceph_demon_counter[demon.role] + increment if ceph_demon_counter.get(
+                demon.role) else increment
+        return ceph_demon_counter
+
+    def get_metadata_list(self, role, client=None):
+        """
+        Returns metadata for demons of specified role
+
+        Args:
+            role(str): ceph demon role
+            client(CephObject): Client with keyring and ceph-common
+
+        Returns:
+            list: metadata as json object representation
+
+        """
+        if not client:
+            client = self.get_ceph_object('client') if self.get_ceph_object('client') else self.get_ceph_object('mon')
+
+        out, err = client.exec_command('sudo ceph {role} metadata -f json-pretty'.format(role=role))
```

Comment:
since `err` is not being used, I suggest calling this var `_` as a throwaway var

Reply:
done